### ISSUE TITLE
pressing F1 can't open https://docs.gimp.org/2.10/en/gimp-help.xml' for reading: Operation not supported

### ISSUE THREAD

**Issue by skierpage** [flathub/org.gimp.GIMP#110]
## Update by @brunvonlope 

If you experience this "bug" install `gvfs` package provided by your package manager. It is a dependency of GIMP that can not be provided by GNOME Runtime since it included daemons.

---

If I press F1 for help, I get
> ⚠ help Message
> Could not open 'https://docs.gimp.org/2.10/en/gimp-help.xml' for reading: Operation not supported
> Perhaps you are missing GIO backends and need to install GVFS?

Gimp upstream is tracking the same issue with its snap in [issue 6515](https://gitlab.gnome.org/GNOME/gimp/-/issues/6515).

This is with app/org.gimp.GIMP/x86_64/stable version 2.10.24 from flathub running on Fedora 34 KDE spin.

Oddly, `flatpak info org.gimp.GIMP --verbose` reports the latest stable(?) commit as:
>   Subject: Issue #101 - Add gvfsd permission for gvfs (f819d561)
but maybe web access needs to be granted too?

**Comment by hfiguiere** [comment]
I have `org.gimp.GIMP.Manual` installed in addition to GIMP and help opens.

Is this still a problem with you?

**Comment by skierpage** [comment]
> I have `org.gimp.GIMP.Manual` installed in addition to GIMP and help opens.
OK, but I don't want to install the manual, I want GIMP to load its help from the web. There is content at that URL, suggesting that some permission or library is missing in the Flatpak.

> Is this still a problem with you?
Yes, it still happens. I'm now on GIMP 2.10.30.

**Comment by Jehan** [comment]
> Gimp upstream is tracking the same issue with its snap in issue 6515.

For the record, the snap is not official (for now, there has been discussions, but nothing ever finalized), unlike the flatpak. We just don't close the report immediately because we also help third-party packages when we can, and it allows us to follow a bit what's happening elsewhere. 🙂

> I have org.gimp.GIMP.Manual installed in addition to GIMP and help opens.

Hmmm… here too, testing on the stable flatpak, the help opens fine whether I choose it to open from the web or from local version (it's a settings in the Preferences; by default, if the local help is installed, I think that's what it should be set to).

And yes, it is supposed to work in both cases. Not sure why it doesn't in your case.

Could you maybe run this command and paste the output please? Our version output does contain a lot of info which can be useful for debugging.

```
flatpak run org.gimp.GIMP//stable -v
```

**Comment by skierpage** [comment]
Thanks for responding! :bow: 
My GIMP Preferences > Help settings showed
> User manual: Use the online version ⚠
> _The user manual is not installed locally._
> Help browser to use: Web browser"

I tried switching the latter to "Gimp help browser" and did not see any change. I reinstalled org.gimp.GIMP.Manual and switched "Use a locally installed copy" and it still didn't work, but I didn't see the alert "Operation not supported Perhaps you are missing GIO backends and need to install GVFS?"

I attach the `-v` output. I also attach the output of `strace --follow-forks -p {PID of gimp-2.10}`; I didn't see anything particularly useful, gimp is finding "/app/share/gimp/2.0/help/en/gimp-help.xml", and forking a subprocess to run `/app/lib/gimp/2.0/plug-ins/help/help`. I don't see any obvious problems: the subprocess is able to open libgvfscommon.so though it finds no /usr/lib/x86_64-linux-gnu/gvfs/module, then it reads a lot of /app/share/gimp/2.0/help/en/gimp-help.xml. I tried entering the flatpak to run `/app/lib/gimp/2.0/plug-ins/help/help` by hand, but I get
> /app/lib/gimp/2.0/plug-ins/help/help is a GIMP plug-in and must be run by GIMP to be used

Let me know if there's anything else I can do to help debug.

[gimp_stable-v.txt](https://github.com/flathub/org.gimp.GIMP/files/7772251/gimp_stable-v.txt)
[gimp_210_local_help_strace.txt](https://github.com/flathub/org.gimp.GIMP/files/7772252/gimp_210_local_help_strace.txt)

**Comment by Jehan** [comment]
>  it still didn't work

What do "still didn't work" mean? You click F1 and no help is opened? An error message? The help browser opens but it's empty? Something else?

Also your version output looks fine, with a recent package (there have been newest now though).

And as you say, the strace shows the `web-browser` fork (pid 18200), then this PID loads the local `/app/share/gimp/2.0/help/en/gimp-image-window.html` file.

So yes, I'd be interested to know what happens then (i.e. what you see when you say "it didn't work").

**Comment by skierpage** [comment]
> What do "still didn't work" mean? You click F1 and no help is opened?

Yes. I updated then ran `flatpak run org.gimp.GIMP//stable` with Edit > Preferences > Help System "Use a locally installed copy" and "Web Browser". I press [F1] in GIMP's empty main window or [F1] and Help > User Manual > menu items, and nothing happens. No alert window, no UI changes, no console output. One time GIMP's main window flickered. Help > GIMP Online > menu items open up in my browser, but [F1] and Help > User Manual > menu items did nothing.

_However,_ I changed Preferences to "Use a locally installed copy" and "GIMP help browser" and now [F1] and Help > User Manual > menu items open the glorious GIMP Help Browser window :tada: , and I see two clients in `xlsclients` output. I thought I tried this set of options before...

"Use the online version" with "GIMP help browser" displays an identical alert except for the first word: 
>  ⚠ help-browser Message
> Could not open 'https://docs.gimp.org/2.10/en/gimp-help.xml' for reading: Operation not supported
> Perhaps you are missing GIO backends and need to install GVFS?

So I have a workaround, I think this is the right matrix:
|                                        | Use the online version | Use a locally installed copy |
| --- | --- | --- |
| **GIMP help  browser** |  "Could not open... GVFS" alert" | works! :+1: |
| **Web browser**            |  "Could not open... GVFS" alert" |  nothing happens |

I wanted to see if `ltrace` would print anything useful but it's not in the `flatpak enter` shell and my system copy has a glibc mismatch.

Thanks so much for your help

**Comment by bgiobbe** [comment]
I just installed GIMP2.10 on MacOS 12.5 Monterey using Homebrew. When I try to get Help -- with Help/Help or Help/User Manual -- I get the error message in  the title of this issue: 

> Could not open 'https://docs.gimp.org/2.10/en/gimp-help.xml' for reading: Operation not supported
> Perhaps you are missing GIO backends and need to install GVFS?

I got the same error message in GIMP2.9 installed with the .dmg. 
I have seen references to installing a separate gimp-help and I downloaded gimp-help-2.10.0.tar.bz2. The INSTALL file said to do ./configure; make; make install, but configure can't find gimp (or a bunch of other things) and make can't find anything to make.

**Comment by brunvonlope** [comment]
Ok, as per the message this is not a bug on GIMP side. gvfs package (which provides gvfsd-http used by gimp help plugin) needs to be installed on host (it is not (and can not) provided by GNOME runtime).

This is true for the AppImage package too.
